// ember-precompiler-brunch/lib/index.js
// Generated by CoffeeScript 1.3.3
(function() {
  var EmberPrecompiler, fs, jsdom;

  fs = require('fs');

  jsdom = require('jsdom');

  module.exports = EmberPrecompiler = (function() {

    EmberPrecompiler.prototype.brunchPlugin = true;

    EmberPrecompiler.prototype.type = 'template';

    EmberPrecompiler.prototype.extension = 'hbs';

    EmberPrecompiler.prototype.precompile = true;

    EmberPrecompiler.prototype.setup = function(config) {
      this.config = config;
      this.window = jsdom.jsdom().createWindow();
      this.window.run(fs.readFileSync(this.config.files.templates.paths.jquery, 'utf8'));
      this.window.run(fs.readFileSync(this.config.files.templates.paths.handlebars, 'utf8'));
      return this.window.run(fs.readFileSync(this.config.files.templates.paths.ember, 'utf8'));
    };

    function EmberPrecompiler(config) {
      var _ref;
      this.config = config;
      if (((_ref = this.config.files.templates) != null ? _ref.paths : void 0) != null) {
        this.setup(this.config);
      }
      null;
    }

    EmberPrecompiler.prototype.compile = function(data, path, callback) {
      var content, error, result;
      if (!(this.window != null)) {
        return callback("files.templates.paths must be set in your config", {});
      }
      try {
        content = this.precompileEmber(data.toString());
        return result = "Ember.TEMPLATES[module.id] = Ember.Handlebars.template(" + content + ");\nmodule.exports = module.id;";
      } catch (e) {
        return error = e;
      } finally {
        callback(error, result);
      }
    };

    EmberPrecompiler.prototype.precompileEmber = function(data) {
      return this.window.Ember.Handlebars.precompile(data.toString()).toString();
    };

    return EmberPrecompiler;

  })();

}).call(this);
